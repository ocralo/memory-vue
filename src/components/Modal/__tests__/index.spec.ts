import {
	vi,
	describe,
	it,
	expect,
	beforeEach,
	afterEach,
	type Mock,
} from 'vitest'
import { useI18n } from 'vue-i18n'
import { shallowMount } from '@vue/test-utils'
import ModalComponent from '@/components/Modal/index.vue'

vi.mock('vue-i18n')

const scrollToSpy = vi.fn()
global.scrollTo = scrollToSpy

describe('Modal', () => {
	let wrapper: any
	beforeEach(() => {
		;(useI18n as Mock).mockReturnValue({
			t: (value: string) => value,
		})
		wrapper = shallowMount(ModalComponent, {
			props: {
				isOpen: false,
			},
		})
	})

	afterEach(() => {
		wrapper.unmount()
		vi.clearAllMocks()
	})

	describe('mount', () => {
		it('should render correctly', () => {
			expect(wrapper.html()).toMatchSnapshot()
		})
	})

	describe('methods', () => {
		describe('handleClose', () => {
			it('should emit close event', () => {
				wrapper.vm.handleClose()
				expect(wrapper.emitted('close')).toBeTruthy()
			})
		})

		describe('handleScroll', () => {
			it('should emit scroll event', () => {
				wrapper.vm.handleScroll(true)
				expect(scrollToSpy).toHaveBeenCalledWith(0, 0)
			})

			it('should not emit scroll event', () => {
				wrapper.vm.handleScroll(false)
				expect(scrollToSpy).not.toHaveBeenCalled()
			})
		})
	})
})
