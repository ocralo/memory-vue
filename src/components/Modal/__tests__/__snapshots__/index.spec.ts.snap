// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Modal > mount > should render correctly 1`] = `
"<div class=\\"container mx-auto\\">
  <div class=\\"flex justify-center\\">
    <div class=\\"absolute inset-0 flex items-center justify-center bg-gray-700 bg-opacity-50 z-10\\" style=\\"display: none;\\">
      <div class=\\"max-w-2xl p-6 bg-white rounded-md shadow-xl\\">
        <div class=\\"flex items-center justify-between\\">
          <div></div><button title=\\"modal.close\\" class=\\"cursor-pointer\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"w-8 h-8 text-slate-700 cursor-pointer\\" fill=\\"none\\" viewBox=\\"0 0 24 24\\" stroke=\\"currentColor\\">
              <path stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" stroke-width=\\"2\\" d=\\"M10 14l2-2m0 0l2-2m-2 2l-2-2m2 2l2 2m7-2a9 9 0 11-18 0 9 9 0 0118 0z\\"></path>
            </svg></button>
        </div>
        <div class=\\"mt-4\\"></div>
      </div>
    </div>
  </div>
</div>"
`;
